src/lib.rs

```diff
@@ -99,33 +124,67 @@ pub fn run(args: Args) {
             );
 
             get_searcher(output_mode)
+                .borrow_mut()
                 .search_path(&matcher, path, printer.sink_with_path(&matcher, path))
                 .unwrap();
+            buffer_writer.print(printer.get_mut()).unwrap();
         });
 }
 
-fn format_relative_path(path: &Path, output_mode: OutputMode) -> &Path {
-    if output_mode == OutputMode::Vimgrep && path.starts_with("./") {
-        path.strip_prefix("./").unwrap()
-    } else {
-        path
-    }
+type Printer = Standard<Buffer>;
+
+thread_local! {
+    static PRINTER: OnceCell<(Rc<RefCell<Printer>>, OutputMode)> = Default::default();
+}
+fn get_printer(buffer_writer: &BufferWriter, output_mode: OutputMode) -> Rc<RefCell<Printer>> {
+    PRINTER.with(|printer| {
+        let (printer, output_mode_when_initialized) = printer.get_or_init(|| {
+            (
+                Rc::new(RefCell::new(create_printer(buffer_writer, output_mode))),
+                output_mode,
+            )
+        });
+        assert!(
+            *output_mode_when_initialized == output_mode,
```

Comment:
Just an "invariant assertion" related to the fact that we're "caching" these printers/searchers but that they're in fact contingent (at the time of creation/caching) on an `OutputMode`